crates/cmd/src/main.rs

```diff
@@ -56,7 +59,45 @@ async fn run() {
         }
     }
 
-    ApiService::run(config).await;
+    let mut handles = Vec::new();
+
+    if !config.router_config.enabled_routes.is_empty() {
+        let api_config = config.clone();
+        let api_handle = tokio::spawn(async move {
+            tracing::info!("Starting API service...");
+            ApiService::run(api_config).await;
+        });
+        handles.push(api_handle);
+    } else {
+        tracing::warn!("No relay API routes are enabled.");
+    }
+
+    if config.website.enabled {
+        let website_config = config.clone();
+        let website_handle = tokio::spawn(async move {
+            loop {
+                tracing::info!("Starting website service...");
+                match WebsiteService::run(website_config.clone()).await {
+                    Ok(_) => {
+                        tracing::error!("Website service unexpectedly completed. Restarting...");
+                    }
+                    Err(e) => {
+                        tracing::error!("Website server error: {}. Restarting...", e);
+                    }
+                }
+                tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+            }
+        });
+        handles.push(website_handle);
+    }
+
+    if !handles.is_empty() {
+        // Await all tasks to prevent the runtime from shutting down
+        futures::future::join_all(handles).await;
+    } else {
+        tracing::error!("No services are enabled.");
+        std::process::exit(1);
+    }
```

Comment:
Can we clean up this logic a bit? Feels a bit cluttered for being the `main` function. Also the semantics of `enabled_routes` might not match with what's done here. Can you double check it?